### ISSUE TITLE
feat request for posthtml-mso

### ISSUE THREAD

**Issue by Scrum** [posthtml/posthtml-render#36]
Need to use it in `posthtml-mso`, so this:

```html
<outlook><table><tr><td></outlook>
```

results in this:

```html
<!--[if mso|ie]><table><tr><td><![endif]-->
```

and not in this:

```html
<!--[if mso|ie]><table><tr><td></td></tr></table><![endif]-->
```

**Comment by cossssmin** [comment]
Additional context:

It's common when coding HTML emails to use 'ghost tables' for Outlook - this is markup written inside conditional comments that target Outlook on Windows, so that it gets used only in that email client.

The most frequent usage is to define fixed-width tables for Outlook when coding mobile-first emails - a technique also known as 'hybrid' in email coding.

The problem with the parser right now is that it automatically closes any unclosed tag it encounters, making it impossible to use [posthtml-mso](https://github.com/posthtml/posthtml-mso) for hybrid emails.

Basically this:

```html
<outlook><table><tr><td></outlook>
  <div>Example</div>
<outlook></td></tr></table></outlook>
```

Is output like this:

```html
<!--[if mso|ie]><table><tr><td></td></tr></table><![endif]-->
    <div>Example</div>

```

Notice also how `</td></tr></table>` doesn't make it in the final output (empty last line).

So the correct output would actually be:

```html
<!--[if mso|ie]><table><tr><td><![endif]-->
    <div>Example</div>
<!--[if mso|ie]></td></tr></table><![endif]-->
```

**Comment by anikethsaha** [comment]
I think this proposal is good. But I strongly belive that the  core posthtml libraries (`posthtml`, `posthtml-parser` and `posthtml-renderer`) should follow the specs as much as possible. also I am not familiar with the `mso` thing . is it a standard thing for html ? (I think its not right ?)

I would prefer to something like [this](https://github.com/posthtml/posthtml/blob/master/lib/index.js#L84-L90) in `posthtml-mso` instead of here cause there might be other plugins which needs custom HTML structure and posthtml 's renderer can be extended to plugins. so I would prefer that. 

let me know. I may miss something.

**Comment by cossssmin** [comment]
> should follow the specs as much as possible

Agree 100%. Not sure how this would be done exactly, but basically what this needs is the unparsed contents of a tag that PostHTML touches - makes sense?

**Comment by anikethsaha** [comment]
> but basically what this needs is the unparsed contents of a tag that PostHTML touches - makes sense?

still a bit lacking. can you give some more details. ?

**Comment by Scrum** [comment]
> I would prefer to something like this in posthtml-mso instead of here cause there might be other plugins which needs custom HTML structure and posthtml 's renderer can be extended to plugins. so I would prefer that.

At the moment, I am of the same opinion since something similar (`Downlevel-revealed` because the parser does not handle correctly) I had to implement in [posthtml-beautify](https://github.com/posthtml/posthtml-beautify)

**Comment by Scrum** [comment]
For now, move it to `posthtml-mso`
